Makefile.inc

```diff
@@ -14,11 +14,22 @@ IDIRS	:=	${IDIRS}
 # Default is no man pages
 MAN	=
 
+# The SUBDIR_DEPTH variable (defined above) initializes tracking the
+# subdirectory depth, but it's not fully set at this stage in the build
+# process.  Instead, we jump back to the shell and print the (finalized)
+# value of SUBDIR_DEPTH.
+FINALIZED_SUBDIR_DEPTH	!=	${MAKE} -v SUBDIR_DEPTH
```

Comment:
I'm not sure how this doesn't result in an infinite loop.  Running `make -V SUBDIR_DEPTH` would process the Makefile and come across this line again, right?

Reply:
Short answer: because `make -v SUBDIR_DEPTH` gets it from `Makefile` (where it's defined in full), not `Makefile.BSD`.

Medium answer: I'll update the comment.

Better answer: hmm, that's weird, `make -f Makefile.BSD -v SUBDIR_DEPTH` and `-V FINALIZED_SUBDIR_DEPTH` both work, instead of going into an infinite loop.  I need to investigate more.

Reply:
Addendum to that "better answer": even if we run `make -f Makefile.BSD -v SUBDIR_DEPTH` on the command-line, the top-level `Makefile.inc` runs `make -v SUBDIR_DEPTH` (which has the fully defined value) so there's no loop.

Reply:
Hmm, so what happens if we run `make Makefiles` and there isn't a Makefile already?  We get a broken one and then it gets fixed the next time we run `make Makefiles`?

Reply:
Not broken.  (Although admittedly, that's a happy accident rather than by design.)

`metabuild.sh` line 25:
```
SUBDIR_DEPTH=$(${MAKEBSD} -v SUBDIR_DEPTH)
```

That works fine regardless of whether there's a `Makefile` or not.

`metabuild.sh` line 172 puts `SUBDIR_DEPTH` into the `Makefile`.  Until that point,
```
.info ${FINALIZED_SUBDIR_DEPTH}
```
in `Makefile.inc` shows that it's empty.  But after line 172 of `metabuild.sh`, it's fine.

Line 179 deals with liball via `copyvar_LIBALL_optional_mutex`.

I've added a comment to `metabuild.sh` to note this dependency.